docs/how-tos/rust_service.md

```diff
@@ -0,0 +1,347 @@
+# Rust Service
+
+## Preface
+
+This guide will explain how to develop a Rust service to serve data to the Graph.
+We will cover:
+
+- Setting up the HTTP/GraphQL Server in [Serving GraphQL Requests](#serving-graphql-requests)
+- Creating GraphQL Objects and providing field resolvers in [Defining Objects](#defining-objects)
+- Extending the graph using [Federation](#federated)
+- Exporting the GraphQL schema for use in federation in [Schema Generation](#schema-generation)
+
+## Dependencies
+
+This guide will utilize the following dependencies:
+
+- `axum` as the HTTP server
+- `async-graphql` as graphql server library 
+- `async-graphql-axum` for GraphQL request handling
+- `tokio` as runtime for writing asynchronous rust program with macros and rt-multi-thread feature enabled
+
+??? tip "More about async-graphql"
+
+    `async-graphql` provides a high-performance server-side implementation of the GraphQL specification.
+    It is designed to leverage Rust's asynchronous programming capabilities (using Tokio or async-std) to handle GraphQL queries efficiently and concurrently.
+
+
+```toml
+[dependencies]
+async-graphql = "7.0.6"
+async-graphql-axum = "7.0.6"
+axum = "0.7.5"
+tokio = { version = "1.38.0", features = ["macros", "rt-multi-thread"] }
+```
+
+## Serving GraphQL Requests
+
+To handle GraphQL requests, we start by creating a server with a placeholder GraphQL schema. We use the `async-graphql-axum` crate, which provides integration between `async-graphql` and `axum`, to create a `graphql_handler`.
```

Comment:
Would be good to mention the registration of a `route` and the `serve`ing of the http API with `axum`